Src/AutoFixtureUnitTest/Utf8EncodingGeneratorTest.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Ploeh.AutoFixture;
+using Ploeh.AutoFixture.Kernel;
+using Ploeh.AutoFixtureUnitTest.Kernel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace Ploeh.AutoFixtureUnitTest
+{
+    public class Utf8EncodingGeneratorTest
+    {
+        [Fact]
+        public void SutIsSpecimenBuilder()
+        {
+            var sut = new Utf8EncodingGenerator();
+            Assert.IsAssignableFrom<ISpecimenBuilder>(sut);
+        }
+
+        [Fact]
+        public void CreateWithEncodingRequestWillReturnUtf8Encoding()
+        {
+            // Arrange
+            var sut = new Utf8EncodingGenerator();
+
+            // Act
+            var result = sut.Create(typeof(Encoding), new DelegatingSpecimenContext());
+
+            // Assert
+            Assert.Equal(Encoding.UTF8, result);
+        }
+
+        [Fact]
+        public void CreateWithNullRequestWillReturnNoSpecimen()
+        {
+            // Arrange
+            var sut = new Utf8EncodingGenerator();
+
+            // Act
+            var result = sut.Create(null, new DelegatingSpecimenContext());
+
+            // Assert
+            Assert.Equal(new NoSpecimen(), result);
+        }
+
+        [Fact]
+        public void CreateWithNullContextDoesNotThrow()
```

Comment:
I was already wondering how to handle something like that. I have seen tests to verify that, but indeed it's not really a requirement, but just a relaxed requirement that might change in the future.

Of course, when you start using that parameter, then you would add a test to also verify that it **does** throw when you give it `null`. So in that situation, you would just modify the test to meet its new requirement.

Would you say that that is a violation of the open/closed principle? I always wonder in what extend something like the OCP applies. Isn't changing the class that it uses the parameter already a violation of OCP, or is that stretching it?

Anyway, I've removed the test case, since I agree that it's not necessary for this class.